File: app/src/main/java/com/example/mvctutorial/designpattern/_08absttractfactory_guifactory/mac/MacButton.java
package com.example.mvctutorial.designpattern._08absttractfactory_guifactory.mac;

import com.example.mvctutorial.designpattern._08absttractfactory_guifactory.Button;

public class MacButton implements Button {
    @Override
    public void onClick() {
        System.out.println("mac button click");
    }
}

File: app/src/main/java/com/example/mvctutorial/designpattern/_08absttractfactory_guifactory/mac/MacButton.kt
package com.example.mvctutorial.designpattern._08absttractfactory_guifactory.mac

import com.example.mvctutorial.designpattern._08absttractfactory_guifactory.Button

class MacButton: Button {
    override fun onClick() {
        println("windows button clicked")
    }
}